guide/event-handling/README.md

```diff
@@ -0,0 +1,284 @@
+if (!message.content.startsWith(prefix) || message.author.bot) return;
++ const client = message.client;
+```
+
+Now run your bot and check whether it's working or not. If you encounter any error, go back and check you haven't made any mistake.
+
+You can add back all the additional features in the `message.js` event file. Just make sure you take care of the things that we talked about above.
+
+## Passing client to event files
+
+You must have noticed how important `Client` class is in discord.js. You created a `client` instance of this class in the `index.js` file. Most of the time you can use this `client` in other files by either obtaining it from one of the `discord.js` structures or from function parameters. You did the former in the above section by using `<Message>.client`. When you don't have access to any of the structures that have `client` as one of their properties, you'll have to use the latter method. The perfect example of this is the `ready.js` event file.
+
+The `ready` event does not return anything, this means that `args` will be an empty array, thus nothing will be passed to the `execute` function in the `ready.js` event file. Now to obtain the `client` you'll have to pass it as an argument along with the `args` array in the event-handler. So, head back to the `index.js` file and make the following changes in it:
+
+```diff
+for (const file of eventFiles) {
+	const event = require(`./events/${file}`);
+	if (event.once) {
+-		client.once(event.name, (...args) => event.execute(...args));
++		client.once(event.name, (...args) => event.execute(...args, client));
+	} else {
+-		client.on(event.name, (...args) => event.execute(...args));
++		client.on(event.name, (...args) => event.execute(...args, client));
+	}
+}
+```
+
+Now, `client` will be available as the last argument to the `execute` function in every event file you'll create. Since, `ready` event doesn't return anything, `client` will be the only argument to the `execute` function of `ready.js` event file. You can make use of `client` in `ready.js` by logging your bot's name in the console when it becomes ready. Head back to `ready.js` and make the following changes to it:
+
+```diff
+module.exports = {
+	name: 'ready',
+    once: true,
++	execute(client) {
+-		console.log('Ready!');
++		console.log(`Logged in as ${client.user.tag}`);
+	},
+};
+```
+
+You can omit the `client` argument from the `execute` function in an event file if you either obtain it from some other discord.js structure or don't need the `client` in it at all. For example, the `message.js` event file will still work if you don't take `client` as an argument in the `execute` function, but if you want you can make the following changes, and it will work just like earlier: 
+
+```diff
+module.exports = {
+	name: 'message',
+	once: false,
++	execute(message, client) {
+		if (!message.content.startsWith(prefix) || message.author.bot) return;
+-		const client = message.client;
+		// ...
+	},
+};
+```
```

Comment:
```suggestion
You can omit the `client` argument from the `execute` function in files where you don't need it. For example, it isn't needed in the `message.js` file because its first argument is a `Message` instance, meaning you can use `message.client`.
```